test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -607,6 +752,42 @@ public void update(Observable obs, Object arg) {
         }
     }
 
+    /**
+     * StreamListener for updates to statusTable, utilizes countdown latch to count entries coming
+     * from markSnapshotSyncInfoOngoing and markSnapshotSyncInfoCompleted.
+     *
+     */
+    class TestStatusTableStreamListener implements StreamListener {
+        @Getter
+        CopyOnWriteArrayList<CorfuStreamEntries> entries = new CopyOnWriteArrayList<>();
+
+        @Getter
+        Throwable throwable;
+
+        CountDownLatch countDownLatch;
+
+        public TestStatusTableStreamListener(CountDownLatch countDownLatch) {
+            this.countDownLatch = countDownLatch;
+        }
+
+        public void onNext(CorfuStreamEntries results) {
+            entries.add(results);
+            for (List<CorfuStreamEntry> entry : results.getEntries().values()) {
+                ReplicationStatusVal status = (ReplicationStatusVal) entry.get(0).getPayload();
+                SyncType syncType = status.getSyncType();
+                SyncStatus syncStatus = status.getSnapshotSyncInfo().getStatus();
+                if (syncType == SyncType.LOG_ENTRY && syncStatus == SyncStatus.COMPLETED ||
+                        syncType == SyncType.SNAPSHOT && syncStatus == SyncStatus.ONGOING) {
```

Comment:
counting down the latch by validating the sync type and status is not required.  In fact, it may also mask some error.  In the tests, we do not expect any other update to be written after init and before entry to snapshot or log entry sync.  So the latch should be counted down here without any checks and the test verifies the update